### PR TITLE
Remove long form Dispose

### LINKED ISSUE #3834: Get rid of 'long form' dispose on MvcForm
MvcForm has the 'long form' dispose implementation that's only really needed if you use native resources and need finalization. We can remove this.

I did a search and there are 3-4 other places that include `GC.SuppressFinalize` include (gasp) Controller and ApiController. None of these have finalizers.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #3834

### CODE REVIEW

**Comment by pranavkm** [comment]
cc @rynowak

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs [hunk 1]
```diff
@@ -8,11 +8,18 @@
 
 namespace Microsoft.AspNet.Mvc.Rendering
 {
+    /// <summary>
+    /// Represents an HTML form element in an MVC view.
```
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs]: Ick "represents". This object represents nothing at all. It's an implementation detail used to ensure we generate one and only one `</form>` end tag.

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs]: I thought @DamianEdwards wanted us to "Represents ..".

Comment (rynowak) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs]: This **does** actually represent an HTML form element.

**Comment by rynowak** [comment]
:shipit:

File: src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs (hunk 1 above)
Comment (dougbu) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs]: Huh?

Comment (DamianEdwards) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs]: I'm mostly concerned with us being consistent, which we aren't, at all. The "Represents..." wording has traditionally been used for .NET API docs.

Comment (pranavkm) [inline src/Microsoft.AspNet.Mvc.ViewFeatures/Rendering/MvcForm.cs]: Ok, I guess we've not been using that term, so I'll keep up with that tradition.

> An HTML form element in an MVC view.

**Pull request closed without merging**